index.js

```diff
@@ -1,300 +1,147 @@
 
-  function convertFormats(type) {
-    const formats = {}
-    for (let i = 0; i < type.formats.length; ++i) {
-      const format = type.formats[i]
-      formats[format.format] = format
-    }
-
-    return { ...type, formats }
-  }
-
-  for (let i = 0; i < TYPES.length; ++i) {
-    const type = TYPES[i]
-    NAMED_TYPES[type.type] = convertFormats(type)
-  }
-
-  return NAMED_TYPES
+const STRING = {
+  TF: Buffer.from([6, 0]),
 }
-
-const NAMED_TYPES = convertTypesToNamedTypes(TYPES)
-
-const FEED = NAMED_TYPES['feed']
-const FEED_T = Buffer.from([FEED.code])
-const CLASSIC_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/classic'].code,
-])
-const GABBYGR_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/gabby-grove'].code,
-])
-const BENDYBT_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/bendy-butt'].code,
-])
-
-const MSG = NAMED_TYPES['msg']
-const MSG_T = Buffer.from([MSG.code])
-const CLASSIC_MSG_TF = Buffer.from([MSG.code, MSG.formats['ssb/classic'].code])
-const GABBYGR_MSG_TF = Buffer.from([
-  MSG.code,
-  MSG.formats['ssb/gabby-grove'].code,
-])
-const BENDYBT_MSG_TF = Buffer.from([
-  MSG.code,
-  MSG.formats['ssb/bendy-butt'].code,
-])
-
-const BLOB = NAMED_TYPES['blob']
-const BLOB_T = Buffer.from([BLOB.code])
-const CLASSIC_BLOB_TF = Buffer.from([
-  BLOB.code,
-  BLOB.formats['ssb/classic'].code,
-])
-
-const SIGNATURE = NAMED_TYPES['signature']
-const SIGNATURE_T = Buffer.from([SIGNATURE.code])
-const SIGNATURE_TF = Buffer.from([
-  SIGNATURE.code,
-  SIGNATURE.formats['ed25519'].code,
-])
-
-const BOX = NAMED_TYPES['encrypted']
-const BOX_T = Buffer.from([BOX.code])
-const BOX1_TF = Buffer.from([BOX.code, BOX.formats['box1'].code])
-const BOX2_TF = Buffer.from([BOX.code, BOX.formats['box2'].code])
-
-const GENERIC = NAMED_TYPES['generic']
-const STRING_TF = Buffer.from([
-  GENERIC.code,
-  GENERIC.formats['UTF8 string'].code,
-])
-const BOOL_TF = Buffer.from([GENERIC.code, GENERIC.formats['boolean'].code])
-const BOOL_TRUE = Buffer.from([1])
-const BOOL_FALSE = Buffer.from([0])
-const NIL_TF = Buffer.from([GENERIC.code, GENERIC.formats['nil'].code])
-const NIL_TFD = NIL_TF
+const BOOL = {
+  TF: Buffer.from([6, 1]),
+  TRUE: Buffer.from([1]),
+  FALSE: Buffer.from([0]),
+}
+const NIL = {
+  TF: Buffer.from([6, 2]),
+}
+NIL.TFD = NIL.TF
 
 const encoder = {
-  feed(feedId) {
-    let tf
-    if (feedId.endsWith('.ed25519')) tf = CLASSIC_FEED_TF
-    else if (feedId.endsWith('.ggfeed-v1')) tf = GABBYGR_FEED_TF
-    else if (feedId.endsWith('.bbfeed-v1')) tf = BENDYBT_FEED_TF
-    else throw new Error('Unknown feed format: ' + feedId)
-
-    const dotIndex = feedId.lastIndexOf('.')
-    const b64part = feedId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+  boolean(input) {
+    const d = input ? BOOL.TRUE : BOOL.FALSE
+    return Buffer.concat([BOOL.TF, d])
   },
+  classic(input, type, format) {
```

Comment:
I keep looking at this name and thinking it has something to do with classic format. At least as long as we still have this kind of classic encoding of other feed formats. Maybe `classicSigilSuffix`? No so good with names.

Reply:
Couple of suggestions:

- `sigilSuffix`
- `ref`
- `sigilRef`

Reply:
DONE